DOCS-10228-dw-format-coercion-example-kt (#1823)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/dataweave-types-coercion.adoc

"08/30/2020"
----

> The following example converts a string to a number with decimal representation. First, you transform the string value `"22"` to `Number` and then to `String` again because `format` adds zeros when converting from a number to a string. If the example string data `"22"` would have been a number you wouldn't need the first `Number` conversion.

Reviewer: ```suggestion
The following example converts a string to a number with decimal representation. First, you transform the string value `"22"` to `Number` and then to `String` again because `format` adds zeros when converting from a number to a string. If the input data is the number `22`, it is not necessary to perform the first `Number` conversion.
```

> The following example converts a string to double. First, you transform the string value `"22"` to `Number` and then to `String` again because `format` add zeros when converting from a number to a string. If the example string data `"22"` would have been a number you wouldn't need the first `Number` conversion.
> The latter `Number` conversion makes the output as a float. Otherwise you would get a string, formatted according to the hosts locale.

Reviewer 2: ```suggestion
The following example converts a string to a number with decimal representation. First, you transform the string value `"22"` to `Number` and then to `String` again because `format` adds zeros when converting from a number to a string. If the example string data `"22"` would have been a number you wouldn't need the first `Number` conversion.
The latter `Number` conversion makes the output a number with the decimal representation. Otherwise you would get a string, formatted according to the hosts locale.
```
Not sure about the last sentence. What is the "hosts locale"? Why is affecting the formatting?
Just changed float and double because in DW these are just numbers with the difference that it has the decimal representation. Just a suggestion, but the script is ok and I understand the explanation
